scylla-cql/src/types/serialize/raw_batch.rs

```diff
@@ -0,0 +1,88 @@
+//! Contains the [`RawBatchValues`] and [`RawBatchValuesIterator`] trait and their
+//! implementations.
+
+use super::row::SerializedValues;
+use super::{RowWriter, SerializationError};
+
+/// A version of `scylla`'s `BatchValues` which knows how to serialize itself
+/// without the need to supply serialization context from the outside.
+pub trait RawBatchValues {
+    /// An `Iterator`-like object over the values from the parent `BatchValues` object.
+    // For some unknown reason, this type, when not resolved to a concrete type for a given async function,
+    // cannot live across await boundaries while maintaining the corresponding future `Send`, unless `'r: 'static`
+    //
+    // See <https://github.com/scylladb/scylla-rust-driver/issues/599> for more details
+    type RawBatchValuesIter<'r>: RawBatchValuesIterator<'r>
+    where
+        Self: 'r;
+
+    /// Returns an iterator over the data contained in this object.
+    fn batch_values_iter(&self) -> Self::RawBatchValuesIter<'_>;
+}
+
+/// An iterator-like for `ValueList`
+///
+/// An instance of this can be easily obtained from `IT: Iterator<Item: ValueList>`: that would be
+/// `BatchValuesIteratorFromIterator<IT>`
+///
+/// It's just essentially making methods from `ValueList` accessible instead of being an actual iterator because of
+/// compiler limitations that would otherwise be very complex to overcome.
+/// (specifically, types being different would require yielding enums for tuple impls)
```

Comment:
In this comment you probably meant `SerializedValues`, not `ValueList`, right?

Reply:
Right, this comment was copied over from the legacy `BatchValuesIterator` but not updated. I'll update in a second.

Reply:
Done